src/lib/components/toggle/index.tsx

```diff
@@ -0,0 +1,136 @@
+import stylin from '@stylin.js/react';
+import { useMotionValue, useSpring } from 'framer-motion';
+import { not } from 'ramda';
+import React, {
+  ChangeEventHandler,
+  FC,
+  PropsWithChildren,
+  useEffect,
+  useState,
+} from 'react';
+
+import { Theme, useTheme } from '../..';
+import { Box, Motion, Typography } from '../../elements';
+import {
+  CheckedButtonElementProps,
+  CheckedButtonProps,
+  LabelElementProps,
+} from './toggle.types';
+
+const CheckedButtonElement = stylin<CheckedButtonElementProps>('input')();
+const LabelElement = stylin<LabelElementProps>('label')();
+
+const TRANSLATE_X = ['0.25rem', '1.3rem'];
+
+export const ToggleButton: FC<PropsWithChildren<CheckedButtonProps>> = ({
+  labels,
+  onChange,
+  disabled,
+  activeIcon,
+  inactiveIcon,
+  defaultValue,
+  ...props
+}) => {
+  const { colors } = useTheme() as Theme;
+  const [toggler, setToggler] = useState(defaultValue || false);
+  const ative = useMotionValue(TRANSLATE_X[Number(toggler)]);
+  const translateX = useSpring(ative, { stiffness: 1000, damping: 100 });
+
+  useEffect(() => {
+    translateX.set(TRANSLATE_X[Number(toggler)]);
+  }, [toggler]);
+
+  const Icon = toggler ? activeIcon : inactiveIcon;
+
+  const handleChange: ChangeEventHandler<HTMLInputElement> = (event) => {
+    onChange?.(event);
+    setToggler(not);
+  };
+
+  return (
+    <Box
+      fontSize="s"
+      display="flex"
+      flexWrap="wrap"
+      fontWeight="300"
+      color="onSurface"
+      alignItems="center"
+      textTransform="capitalize"
+    >
+      <LabelElement
+        ml="0.375rem"
+        display="flex"
+        borderRadius="full"
+        position="relative"
+        nActive={
+          !disabled && {
+            boxShadow: `0 0 0 .25rem ${
+              colors[toggler ? 'primary' : 'onSurface']
+            }29`,
+          }
+        }
+      >
+        <CheckedButtonElement
+          display="none"
+          type="checkbox"
+          checked={toggler}
+          disabled={disabled}
+          onChange={handleChange}
+          {...props}
+        />
+        <Box
+          display="flex"
+          width="2.75rem"
+          height="1.7rem"
+          cursor="pointer"
+          borderRadius="full"
+          alignItems="center"
+          opacity={disabled ? 0.32 : 1}
+          bg={
+            disabled
+              ? `${colors['onSurface']}3D`
+              : !toggler
+              ? `${colors['onSurface']}3D`
+              : 'primary'
+          }
+          nHover={{
+            backgroundImage: toggler
+              ? 'linear-gradient(to bottom, rgba(0, 0, 0, 0.16), rgba(0, 0, 0, 0.16))'
```

Comment:
we need to follow one pattern, plz converter rgba to hex

Reply:
Check if this gradient exists in gradient variable

Reply:
> Check if this gradient exists in gradient variable

We don't have this this gradient, I'm just applying an overlay above this component bg